File: employeetracker/src/main/java/spsoftware/employeetracker/DTO/RegisterRequest.java
package spsoftware.employeetracker.DTO;

import jakarta.persistence.Column;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class RegisterRequest {
    @NotBlank @Email
    public String email;

    @NotBlank
    public String password;

    public String firstName;
    public String lastName;
    public String department;
    public String role;
    public String title;
    public String salary;
    public String phone;
}

File: employeetracker-ui/src/app/models/register-request.ts
export class RegisterRequest {
  email: string;
  password: string;
  firstName?: string;
  lastName?: string;
  department?: string;
  role?: string;
  title?: string;
  salary?: string;
  phone?: string;

  constructor(
    email: string = '',
    password: string = '',
    firstName: string = '',
    lastName: string = '',
    department: string = '',
    role: string = '',
    title: string = '',
    salary: string = '',
    phone: string = '',
  ) {
    this.email = email;
    this.password = password;
    this.firstName = firstName;
    this.lastName = lastName;
    this.department = department;
    this.role = role;
    this.title = title;
    this.salary = salary;
    this.phone = phone;
  }
}
